lib/S3Plugin.js

```diff
@@ -361,6 +412,133 @@ class S3Plugin {
     return stringParam;
   }
 
+  /**
+   * validate given string to match
+   * AWS S3 bucket name format prerequisites
+   */
+    _validateBucketName (name, disableDotsInName=false) {
+      const bucketNameRegex = disableDotsInName ? 
+        // avoiding dots is needed for S3 Transfer Acceleration
+        /(?!(^xn--|.+-s3alias$))^[a-z0-9][a-z0-9-]{1,61}[a-z0-9]$/
+        : /(?!(^((2(5[0-5]|[0-4][0-9])|[01]?[0-9]{1,2})\.){3}(2(5[0-5]|[0-4][0-9])|[01]?[0-9]{1,2})$|^xn--|.+-s3alias$))^[a-z0-9][a-z0-9.-]{1,61}[a-z0-9]$/
+      
+        const sanitizedName = bucketNameRegex.test(name)
+  
+      if (sanitizedName === null) {
+        throw new this.context.errors.BadRequestError(`Invalid bucket name format: ${name}`)
+      }
+      
+      return true
+    }
+
+  /**
+   * Creates a new S3 bucket.
+   *
+   * @controller            bucket
+   * @action                create
+   *
+   * @param {KuzzleRequest} request
+   */
+  async bucketCreate (request) {
+    this._assertCredentials();
+
+    const
+      bucketName = this._stringArg(request, 'bucketName', this.config.bucketName),
+      bucketRegion = this._stringArg(request, 'bucketRegion', this.config.region),
+      disableDotsInName = request.input.body.disableDotsInName,
+      bucketPolicy = request.input.body.bucketPolicy,
+      defaultCORS = {
```

Comment:
Default CORS should be a top level constant

Reply:
done